src/EnergyPlus/api/datatransfer.py

```diff
@@ -84,6 +84,33 @@ class DataExchange:
     outside calling code.
     """
 
+    class _APIDataEntry(Structure):
```

Comment:
This hidden class is basically just a mapping to the C struct.  I understand there is some repetition here, but honestly, once this PR is merged, this stuff probably won't change for years.  I thought about exposing this as the user-facing Python response, but it's just not a nice look.  I think it's much nicer to create a proper Python class to return through the Python API.  Much more polished.